Providers/Scripts/2.4x-2.5x/Scripts/nxOMSPerfCounter.py

```diff
@@ -200,14 +212,13 @@ def ReadOMSAgentConf(HeartbeatIntervalSeconds, PerfCounterObject):
 
 
 def UpdateOMSAgentConf(WorkspaceID, HeartbeatIntervalSeconds, PerfCounterObject):
-    if os.path.exists(conf_path):
-        txt = codecs.open(conf_path, 'r', 'utf8').read().encode(
-            'ascii', 'ignore')
-        LG().Log('INFO', 'Read omsagent configuration ' + conf_path + '.')
-    else:
-        LG().Log(
-            'INFO', 'No omsagent configuration file present.  Will create new  configuration file at ' + conf_path + '.')
-        txt = ''
+    """
+    Write the new values given by parameters to the OMSAgent conf file
+    """
+    txt = get_conf_path_text()
+    if txt == '':
```

Comment:
Why not use an elegant way to refer to empty string in python?
like string.empty, then do 'if not mystring'?

Reply:
The `get_conf_path_text` method returns either an empty string or a non-empty string, so I wanted to be equally explicit in my check for its value. Given this, there is no more elegant way to do it; see https://stackoverflow.com/questions/9573244/most-elegant-way-to-check-if-the-string-is-empty-in-python

Reply:
Yes, that is the point.

Reply:
I've kept the helper method the same (so it will always return a string type), but I've updated the checks to "not txt"